### PR TITLE
HPCC-23209 Add Azure support libraries to the platform

### PR DESCRIPTION
**Pull request by ghalliday**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by ghalliday** [comment]
Not ready for merging.  Needs rebasing, pushed for smoketest

**Comment by ghalliday** [comment]
The option is currently disabled in commonSetup because it requires various boost libraries to be installed as dependencies.

@Michael-Gardner please review.
The necessary libraries build, but if I then make modifications to the source they do not rebuild.  I suspect it may also need to configure where the generated files are placed at install time.  Any comments very welcome.

File: cmake_modules/commonSetup.cmake [hunk 1]
```diff
@@ -66,6 +66,7 @@ IF ("${COMMONSETUP_DONE}" STREQUAL "")
   option(USE_OPENSSL "Configure use of OpenSSL" ON)
   option(USE_ZLIB "Configure use of zlib" ON)
   option(USE_CBLAS "Configure use of cblas" ON)
+  option(USE_AZURE "Configure use of azure" ONF)
```
Comment (richardkchapman) [inline cmake_modules/commonSetup.cmake]: ONF ?

Comment (ghalliday) [reply cmake_modules/commonSetup.cmake]: A sort of half way house!

**Comment by ghalliday** [comment]
Richard pointed out it might be better to use https://github.com/Azure/azure-storage-cpplite which is lighter weight.  Should take a close look at the differences.

**Pull request closed without merging**